packages/gatsby/src/redux/reducers/__tests__/redirects.js

```diff
@@ -71,4 +71,23 @@ describe(`redirects`, () => {
       ])
     })
   })
+
+  it(`prevents duplicate redirects`, () => {
+    function createRedirect(fromPath, toPath) {
+      return {
+        type: `CREATE_REDIRECT`,
+        payload: { fromPath, toPath },
+      }
+    }
+
+    let state = reducer(undefined, createRedirect(`/page`, `/other-page`))
+    state = reducer(state, createRedirect(`/page`, `/other-page`))
+
+    expect(state).toEqual([
+      {
+        fromPath: `/page`,
+        toPath: `/other-page`,
+      },
+    ])
+  })
```

Comment:
Could we add `allows multiple redirects with same "fromPath" but different options` test case to catch potential regressions in the future?

Reply:
Done. Please take another look.